### PR TITLE
Renaming of files from eXstatic

### PR DESCRIPTION
**Pull request by saurabh-s-sawant**
-Renamed original CodeUtils folder from eXstatic (which was named FerroXUtils) to eXstaticUtils along with the corresponding *.H and *.cpp files.

-Renamed MultiFab_Manipulation namespace to eXstatic_MFab_Utils.

-Separated FerroX specific Utils into the folder FerroXUtils. 

-Created Ferrox_MFab_Utils namespace which will contain FerroX specific MFab utils. 

-For a file taken from eXstatic, a comment has been added at the top stating that it is taken from eXstatic.

-For a file taken from eXstatic and modified in FerroX, a comment has been added stating that 
it is taken from eXstatic and has been modified for FerroX.

### CODE REVIEW

**Comment by prkkumar** [comment]
Thanks for this PR @PeculiarOvertones! Looks good to me. I ran the https://github.com/AMReX-Microelectronics/FerroX/blob/development/Exec/Examples/inputs_mfim_Noeb input and I am getting the same results as before.

**Review by prkkumar — APPROVED** [review]

**Comment by saurabh-s-sawant** [comment]
@prkkumar  Thank you for checking!

File: Source/Solver/ElectrostaticSolver.cpp [hunk 1]
```diff
@@ -485,9 +485,9 @@ void Fill_FunctionBased_Inhomogeneous_Boundaries(c_FerroX& rFerroX, MultiFab& Po
                         [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
                         {
 		            #ifdef TIME_DEPENDENT
-                                Multifab_Manipulation::ConvertParserIntoMultiFab_4vars(i,j,k,time,dx,real_box,iv,macro_parser,soln_arr);
+                                eXstatic_MFab_Util::ConvertParserIntoMultiFab_4vars(i,j,k,time,dx,real_box,iv,macro_parser,soln_arr);
 		            #else
-                                Multifab_Manipulation::ConvertParserIntoMultiFab_3vars(i,j,k,dx,real_box,iv,macro_parser,soln_arr);
+                                eXstatic_MFab_Util::ConvertParserIntoMultiFab_3vars(i,j,k,dx,real_box,iv,macro_parser,soln_arr);
                             #endif
                             if(i == 5 && j == 5 && k == 65) amrex::Print() << "time = " << time << ", phi = " << soln_arr(i,j,k) << "\n";
```
Comment (ajnonaka) [inline Source/Solver/ElectrostaticSolver.cpp]: amrex::Print() doesn't compile/work on the GPU inside of a ParallelFor.  Either comment this out or change to printf().

**Review by ajnonaka — APPROVED** [review]
Approve, pending the one comment I added

File: Source/Solver/ElectrostaticSolver.cpp (hunk 1 above)
Comment (prkkumar) [reply Source/Solver/ElectrostaticSolver.cpp]: Thanks, @ajnonaka ! I removed that comment (wasn't needed).

**Pull request merged**